client/test/client.go

```diff
@@ -66,23 +67,37 @@ func (c *Client) UpdateLastBlockNum() {
 // Deposit deposits amount to the Erdstall contract and waits until the deposit
 // tx is mined. The minedBlockNum is updated automatically.
 func (c *Client) Deposit(ctx context.Context, amount *big.Int) error {
-	tr, err := c.ethClient.NewTransactor(ctx, amount, eth.DefaultGasLimit, c.ethClient.Account())
-	if err != nil {
-		return fmt.Errorf("creating transactor: %w", err)
-	}
-
-	tx, err := c.contract.Deposit(tr)
-	if err != nil {
-		return fmt.Errorf("calling deposit: %w", err)
-	}
+	return c.call(ctx, func(ctx context.Context) (*types.Transaction, error) {
+		tr, err := c.ethClient.NewTransactor(ctx, amount, eth.DefaultGasLimit, c.ethClient.Account())
```

Comment:
I opted for creating the transactor in each call (like you also suggested) but just in case this question might come up:

One could also dedup the transactor creation by letting the caller pass an extra `amount` to `Client.call(...)` but this broke one too many logical barriers imo, so I also settled for the current impl.

Reply:
I just remembered that we don't set the contexts yet in `NewTransactor`, so everywhere where you create the `tr`, after the `err != nil` check, do `tr.Context = ctx`. It will be fixed in go-perun soon.

Even better yet, you could also create the method `NewTransactor` on the eth client and do it there, then we only need to change it in one place.

Reply:
And go-perun will also soon remove the `amount` parameter, then it needs to be set manually using `tr.Amount = amount` anyways. So I think you can change the function passed to `call` to `func(context.Context, tr *bind.TransactOpts) (*types.Transaction, error)` and then set the amount on `tr` in `Deposit`.